common/src/cell.rs

```diff
@@ -266,3 +266,43 @@ impl<'a, T: ?Sized> PyMappedRwLockWriteGuard<'a, T> {
         PyMappedRwLockWriteGuard(MappedRwLockWriteInner::map(s.0, f))
     }
 }
+
+struct StaticCellInner<T> {
+    value: T,
+}
+
+// SAFETY: Inner value must be Send+Sync when threading is on; Otherwise single-thread.
+#[cfg(not(feature = "threading"))]
+unsafe impl<T> Send for StaticCellInner<T> {}
+#[cfg(not(feature = "threading"))]
+unsafe impl<T> Sync for StaticCellInner<T> {}
+
+pub struct StaticCell<T> {
+    inner: ::once_cell::sync::OnceCell<StaticCellInner<T>>,
```

Comment:
I'll take a better look at this tomorrow, but maybe this could be like a `OnceCell<T>` when threading but a `&'static std::thread::LocalKey<T>` otherwise? I don't like the look of that `unsafe impl Send + Sync` block...

Reply:
That will be possible but sounds like to have different implementation for StaticCell for threading and non threading.